clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -985,6 +985,38 @@ def CmpOp : CIR_Op<"cmp", [Pure, SameTypeOperands]> {
   let hasVerifier = 0;
 }
 
+//===----------------------------------------------------------------------===//
+// ThreeWayCmpOp
+//===----------------------------------------------------------------------===//
+
+def ThreeWayCmpOp : CIR_Op<"three_way_cmp", [Pure, SameTypeOperands]> {
+  let summary = "Compare two values with C++ three-way comparison semantics";
+  let description = [{
+    The `cir.three_way_cmp` operation models the `<=>` operator in C++20. It
+    takes two operands with the same type and produces a result indicating the
+    ordering between the two input operands.
+
+    The result of the operation is a signed integer that indicates the ordering
+    between the two input operands.
```

Comment:
Is this signed integer supposed to be of any specific size? You should make the result constrained at least to be signed (for instance, if it's always s8i, then you could use `SInt8` constraint).

Reply:
The width of the result type cannot be fixed, it depends on the standard library implementation to avoid unnecessary integer casts during CIRGen. The result type do need to be a signed type; I have added verification for this in CIRDialect.cpp.